packages/tenant-process/src/services/tenantService.ts

```diff
@@ -69,38 +70,72 @@ export function tenantServiceBuilder(
       attributeId: AttributeId,
       verifierId: string
     ): Promise<string> {
-      const tenant = await readModelService.getTenantById(tenantId);
+      logger.info(
+        `Update extension date of attribute ${attributeId} for tenant ${tenantId}`
+      );
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
 
-      return await repository.createEvent(
-        await updateVerifiedAttributeExtensionDateLogic({
-          tenantId,
-          attributeId,
-          verifierId,
-          tenant,
-        })
+      const attribute = tenant?.data.attributes.find(
```

Comment:
this cannot be `undefined` now, right?
```suggestion
      const attribute = tenant.data.attributes.find(
```